daemons/controld/controld_callbacks.c

```diff
@@ -171,6 +171,17 @@ peer_update_callback(enum crm_status_type type, crm_node_t * node, const void *d
 
             if (!appeared) {
                 controld_remove_voter(node->uname);
+            } else if (!AM_I_DC && !is_remote) {
+                /*
+                 * This is a hack until we can send to a nodeid and/or we fix node name lookups
+                 * These messages are ignored in crmd_ha_msg_filter()
+                 */
+                xmlNode *query = create_request(PCMK__CONTROLD_CMD_POKE, NULL, NULL, CRM_SYSTEM_CRMD, CRM_SYSTEM_CRMD, NULL);
+
+                crm_debug("Broadcasting our uname because of node %u", node->id);
+                send_cluster_message(node, crm_msg_crmd, query, FALSE);
+
+                free_xml(query);
```

Comment:
I've just realized we can simplify this even more. There's already a CRM_OP_HELLO that is ignored (in relay_message()). Currently, CRM_OP_HELLO is only sent via IPC, but this would be a good use of it for CPG. Just replace the POKE with the HELLO here, and the rest of the PR can be done away with.

That avoids needing a special case in crmd_cs_dispatch(), and makes sure the message goes through all the usual handling.

Reply:
> I've just realized we can simplify this even more. There's already a CRM_OP_HELLO that is ignored (in relay_message()). Currently, CRM_OP_HELLO is only sent via IPC, but this would be a good use of it for CPG. Just replace the POKE with the HELLO here, and the rest of the PR can be done away with.
> 
> That avoids needing a special case in crmd_cs_dispatch(), and makes sure the message goes through all the usual handling.

Thanks, done!